resolver-dns/src/main/java/io/netty/resolver/dns/DnsServerAddressStreamProvider.java

```diff
@@ -25,12 +25,29 @@
  * /etc/resolver</a>.
  */
 @UnstableApi
-public interface DnsServerAddressStreamProvider {
+public abstract class DnsServerAddressStreamProvider {
+    // TODO(scott): how is this done on Windows? This may require a JNI call to GetNetworkParams
+    // https://msdn.microsoft.com/en-us/library/aa365968(VS.85).aspx.
+    private static final DnsServerAddressStreamProvider DEFAULT_DNS_SERVER_ADDRESS_STREAM_PROVIDER =
+            UnixResolverDnsServerAddressStreamProvider.parseSilently();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Initialization of io.netty.resolver.dns.DnsServerAddressStreamProvider accesses class io.netty.resolver.dns.UnixResolverDnsServerAddressStreamProvider, which isn't initialized yet [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3AIC_SUPERCLASS_USES_SUBCLASS_DURING_INITIALIZATION)